src/state.ts

```diff
@@ -570,6 +572,35 @@ export default class State {
         this._activities[rsn][activity] = score;
     }
 
+    hasGuildSettings(guildId: string): boolean {
```

Comment:
I'm not sure if the settings should have the same getters and setters as the boss/skill/clue data. I think we just need one getter that safely does a check on the state and returns the default if it's not present. I say this because we're not doing any checks or computing diffs.

Plus we need to ensure it's wiped when the bot is deleted from a guild.